packages/odyssey-react-mui/src/Typography.tsx

```diff
@@ -0,0 +1,334 @@
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import {
+  Typography as MuiTypography,
+  TypographyProps as MuiTypographyProps,
+} from "@mui/material";
+import { ElementType, ReactNode, useMemo } from "react";
+
+export type TypographyVariantValue =
+  | "h1"
+  | "h2"
+  | "h3"
+  | "h4"
+  | "h5"
+  | "h6"
+  | "body"
+  | "subtitle2"
+  | "caption"
+  | "legend";
+
+export const typographyVariantMapping: Record<
+  TypographyVariantValue,
+  MuiTypographyProps["variant"]
+> = {
+  h1: "h1",
+  h2: "h2",
+  h3: "h3",
+  h4: "h4",
+  h5: "h5",
+  h6: "h6",
+  body: "body1",
+  caption: "subtitle1",
+  subtitle2: "subtitle2",
+  legend: "legend",
+} as const;
+
+export const typographyColorValues = [
+  "primary",
+  "textPrimary",
+  "secondary",
+  "textSecondary",
+  "error",
+] as const;
+
+export type TypographyProps = {
+  /**
+   * The ID of the element that describes the component.
+   */
+  ariaDescribedBy?: string;
+  /**
+   * The ARIA label for the component.
+   */
+  ariaLabel?: string;
+  /**
+   * The ID of the element that labels the component.
+   */
+  ariaLabelledBy?: string;
+  /**
+   * The text content of the component.
+   */
+  children: ReactNode;
+  /**
+   * Additional classes to add to the component.
+   */
+  classes?: object;
+  /**
+   * The color of the text.
+   */
+  color?: (typeof typographyColorValues)[number];
+  /**
+   * The HTML element the component should render, if different from the default.
+   */
+  component?: ElementType;
+  /**
+   * The variant of Typography to render.
+   */
+  variant?: keyof typeof typographyVariantMapping;
+};
+
+export const Typography = ({
+  ariaDescribedBy,
+  ariaLabel,
+  ariaLabelledBy,
+  children,
+  classes,
+  color,
+  component: componentProp,
+  variant = "body",
+}: TypographyProps) => {
+  const component = useMemo(() => {
+    if (!componentProp) {
+      if (variant === "body") {
+        return "p";
+      } else if (variant === "caption" || variant === "subtitle2") {
+        return "h6";
+      } else {
+        return variant;
+      }
+    }
+    return componentProp;
+  }, [componentProp, variant]);
+
+  return (
+    <MuiTypography
+      ariaDescribedBy={ariaDescribedBy}
+      ariaLabel={ariaLabel}
+      ariaLabelledBy={ariaLabelledBy}
+      children={children}
+      classes={classes}
+      color={color}
+      component={component}
+      variant={typographyVariantMapping[variant]}
+    />
+  );
+};
+
+Typography.displayName = "Typography";
+
+export const H1 = ({
+  ariaDescribedBy,
+  ariaLabel,
+  ariaLabelledBy,
+  children,
+  classes,
+  color,
+  component,
+}: TypographyProps) => (
+  <Typography
+    ariaDescribedBy={ariaDescribedBy}
+    ariaLabel={ariaLabel}
+    ariaLabelledBy={ariaLabelledBy}
+    children={children}
+    classes={classes}
+    color={color}
+    component={component}
+    variant="h1"
+  />
+);
```

Comment:
What would you say to renaming these?

So this would be `Heading1`, and then `Paragraph`. That's it. I think that would be easier to read. HTML was meant to be smaller in byte-size because we didn't have magnification or compression back then.

Where we can, we should prefer verbose naming.